### DISCUSSION TITLE
AWS Lambda: python-oracledb thin mode cannot be used because the cryptography package is not installed

### DISCUSSION THREAD

**Discussion by ghost** [oracle/python-oracledb#68] [General]
Creating a new thread on this particular issue because the error and issue are deceiving.

The installation instructions say: `python -m pip install oracledb`

This I've done and I packaged this into a lambda layer. I am able to 
```
import oracledb
connection = oracledb.connect(user="me",
                              password="passw1",
                              dsn="myhost_and_service")
```
And this is where I get the error.  Below you can see the cryptography folder/package is installed just like oracledb one.
![image](https://user-images.githubusercontent.com/16198534/192910697-a68f5824-dfe2-4db3-9535-cbaa768e0f5b.png)

Why do I get the error? what must I do to "install" and "configure" the cryptography package? Is there something that needs to be added to the instructions ?

**Comment by cjbj** [comment]
Can any Lambda users help @github4es here?

**Comment by anthony-tuininga** [comment]
I presume you are getting this error?

```
DPY-3016: python-oracledb thin mode cannot be used because the cryptography package is not installed
```

This occurs if the following imports fail:

```python
from cryptography import x509
from cryptography.hazmat.primitives import hashes, serialization
from cryptography.hazmat.primitives.ciphers import algorithms, modes, Cipher
from cryptography.hazmat.primitives.asymmetric import padding
from cryptography.hazmat.primitives.kdf import pbkdf2
```

You could try putting these imports directly in your main script so you can see which one it is complaining about. That may shed some light on the matter?

**Reply by ybindu** [reply]
<img width="653" alt="image" src="https://user-images.githubusercontent.com/108888177/195702802-6d53180c-8252-4fa3-b35b-1d8d3a479460.png">
@anthony-tuininga This is the error I'm getting after importing above suggested packages in my main script.

**Reply by anthony-tuininga** [reply]
See the link below from @davjones12. That seems to be the source of the issue!

**Reply by cjbj** [reply]
@ybindu screenshots aren't great since they aren't (among many things) searchable, so let me add the error from your post in text for future searchability:

```
Unable to import module 'lambda_function': /lib64/libc.so.6: version `GLIBC_2.28' not found (required by /var/task/cryptography/hazmat/bindings/_rust.abi3.so)
```

The solution link kindly provided by @davjones12 had a similar message:

```
Runtime.ImportModuleError: Unable to import module 'main': /lib64/libc.so.6: version `GLIBC_2.18' not found (required by /var/task/cryptography/hazmat/bindings/_rust.abi3.so)
```

**Comment by ybindu** [comment]
Is this issue resolved? If so could someone help on resolving this. I'm also facing same issue.

**Reply by ghost** [reply]
I couldn't get it resolved. So gave up on it because this is for a production system so no time to 'try and fail". Hopefully someone will help you out.

**Reply by davjones12** [reply]
[https://github.com/pyca/cryptography/issues/6391](https://github.com/pyca/cryptography/issues/6391)
